src/LoopbackModelBase.js

```diff
@@ -0,0 +1,103 @@
+const MicroserviceError = require('./MicroserviceError');
+
+module.exports = class LoopbackModelBase {
+    constructor({ modelName, model }) {
+        if (!model) {
+            throw LoopbackModelBase.createError(`No Loopback Model Instance was given to the
+                LoopbackModelBase constructor`);
+        }
+
+        this.modelName = modelName || model.modelName;
+        this[this.modelName] = model;
+
+
+        const functions = this.getPrototypeFunctionsRecursive(this);
+        functions.forEach((prototypeFunction, key) => {
+            if (!this[this.modelName][key]) {
+                this[this.modelName][key] = prototypeFunction.bind(this);
+            } else {
+                throw LoopbackModelBase.createError(`You are overwriting an loopback internal function ${key} on the model ${this.modelName}`);
+            }
+        });
+    }
+    
+    /**
+     * loops recursively trought the given objects pprototypes, reurns all
+     * prototoype functions except 'constructor' and the JavaScript Objects
+     * constructor prototype functions
+     *
+     * @param  {Object} object                  The object to extract prototype functions
+     * @param  {Map}    [functions=new Map()]   Map to store the functions
+     * @return {Map}                            A map with all prototpye functions
+     */
+    getPrototypeFunctionsRecursive(object, functions = new Map()) {
+        const prototype = Object.getPrototypeOf(object);
+        const isLastPrototype = !Object.getPrototypeOf(prototype);
```

Comment:
Originally is was thinking of checking the name of the class, which is available under `instance.constructor.name` but i didn't remember where to find this property. The current solution works fine though.